app/src/org/commcare/dalvik/activities/SingleAppManagerActivity.java

```diff
@@ -0,0 +1,285 @@
+                SharedPreferences preferences = CommCareApplication._().getCurrentApp().getAppPreferences();
+                preferences.edit().putBoolean(CommCarePreferences.AUTO_TRIGGER_UPDATE,true);
+            }
+            break;
+        case CommCareHomeActivity.MISSING_MEDIA_ACTIVITY:
+            refresh();
+            if (resultCode == RESULT_CANCELED) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(this);
+                builder.setTitle("Media Not Verified");
+                builder.setMessage(R.string.skipped_verification_warning_2)
+                    .setPositiveButton("OK", new DialogInterface.OnClickListener() {
+
+                            @Override
+                            public void onClick(DialogInterface dialog,
+                                    int which) {
+                                dialog.dismiss();
+                            }
+                            
+                        });
+                dialog = builder.create();
+                dialog.show();
+            }
+            else if (resultCode == RESULT_OK) {
+                Toast.makeText(this, "Media Validated!", Toast.LENGTH_LONG).show();
+            }
+            break;
+        case CommCareHomeActivity.RESTART_APP:
+            if (dialog != null) {
+                dialog.dismiss();
+            }
+            Intent i = new Intent(getApplicationContext(), AppManagerActivity.class);
+            startActivity(i);
+        }
+    }
+    
+    /** Uninstalls the selected app **/
+    public void uninstall() {
+        CommCareApplication._().logout();
+        CommCareApplication._().initializeAppResources(new CommCareApp(appRecord));
+        CommCareApp app = CommCareApplication._().getCurrentApp();
+        
+        //1) Teardown the sandbox for this app
+        app.teardownSandbox();
+        //2) Delete all the user databases associated with this app
+        SqlStorage<UserKeyRecord> userDatabase = CommCareApplication._().getAppStorage(UserKeyRecord.class);
+        for (UserKeyRecord user : userDatabase) {
+            this.getDatabasePath(CommCareUserOpenHelper.getDbName(user.getUuid())).delete();
+        }
+        //3) Delete the app database
+        this.getDatabasePath(DatabaseAppOpenHelper.getDbName(app.getAppRecord().getApplicationId())).delete();
+        //4) Delete the app record
+        CommCareApplication._().getGlobalStorage(ApplicationRecord.class).remove(appRecord.getID());
+        //5) Reset the appResourceState in CCApplication
+        CommCareApplication._().setAppResourceState(CommCareApplication.STATE_UNINSTALLED);
+        
+        rebootCommCare();
+   }
+    
+    /** If the app is not archived, sets it to archived (i.e. still installed but 
+     * not visible to users); If it is archived, sets it to unarchived **/
+    public void toggleArchived(View v) {
+        appRecord.setArchiveStatus(!appRecord.isArchived());
+        Button b = (Button) v;
```

Comment:
probably best to separate the "Update this button to reflect the current state" since you do it at least twice. Then this method becomes

setArchiveStatus()
write(record)
refresh() (or you refresh outside of this method even) 

that way things stay in sync betweent his method and the original stuff in the oncreate